src/Generation/Generator/Model/Method.cs

```diff
@@ -84,4 +85,80 @@ public static bool IsValidFreeFunction(GirModel.Method method)
     {
         return !method.Parameters.Any() && method.ReturnType.AnyType.TryPickT0(out var type, out _) && type is GirModel.Void;
     }
+
+    /// <summary>
+    /// Whether this method hides a method with the same name and parameters from a parent class
+    /// </summary>
+    public static bool HidesMethod(GirModel.Method method)
+    {
+        if (method.Parent is not GirModel.Class cls)
+        {
+            return false;
+        }
+
+        var publicName = GetPublicName(method);
+        return HidesMethod(cls.Parent, method, publicName);
+    }
+
+    private static bool HidesMethod(GirModel.Class? cls, GirModel.Method method, string publicName)
+    {
+        if (cls is null)
+        {
+            return HidesObjectMethod(method, publicName);
+        }
+
+        var matchingMethod = cls.Methods.FirstOrDefault(m => GetPublicName(m) == publicName);
+
+        if (matchingMethod is null)
+        {
+            return HidesMethod(cls.Parent, method, publicName);
+        }
+
+        GirModel.Parameter[] parameters = method.Parameters.ToArray();
+        GirModel.Parameter[] foundParameters = matchingMethod.Parameters.ToArray();
+
+        if (parameters.Length != foundParameters.Length)
+        {
+            return HidesMethod(cls.Parent, method, publicName);
+        }
+
+        for (var i = 0; i < parameters.Length; i++)
+        {
+            if (!parameters[i].AnyTypeOrVarArgs.Equals(foundParameters[i].AnyTypeOrVarArgs))
+            {
+                return HidesMethod(cls.Parent, method, publicName);
+            }
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Whether this method hides a method from System.Object
+    /// </summary>
+    private static bool HidesObjectMethod(GirModel.Method method, string publicName)
+    {
+        if (method.Parameters.Any())
+        {
+            // We do not currently support detecting overrides of object methods that accept parameters
```

Comment:
It looks like it would be a bit more complicated to check the parameter types against the ones from reflection of the `object` type, and given I've only seen this be an issue for `ToString`, which doesn't take any parameters, it doesn't seem worth the effort to implement this yet.

Reply:
This is fine for now 👍